drivers/net/native_sim_errno.h

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2023 Marcin Niestroj
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __DRIVERS_NET_NATIVE_SIM_ERRNO_H__
+#define __DRIVERS_NET_NATIVE_SIM_ERRNO_H__
+
+#include <errno.h>
+
+#define NSIM_EPERM 1         /**< Not owner */
```

Comment:
Can't we say here
```
#define NSIM_EPERM EPERM
....
```

Reply:
Similar as described in https://github.com/zephyrproject-rtos/zephyr/pull/65116#discussion_r1391146480, `NSIM_*` values are defined to bridge between Native Simulator and Zephyr libc implementations. If we would define `#define NSIM_EPERM EPERM`, `NSIM_EPERM` value would be potentially different in Zephyr, when there is embedded libc used, and different in Native Simulator when there is host libc used. So by defining `#define NSIM_EPERM 1` we make sure that the value is same in those two contexts.

Reply:
Hmm, don't we need to just convert the errno values between zephyr <-> host, so in either zephyr or in host, meaning we could use the errno value defined in one of them?